internal/pkg/manifest/env_test.go

```diff
@@ -509,6 +504,103 @@ http:
 				},
 			},
 		},
+		"unmarshal with new http fields": {
+			inContent: `name: prod
+type: Environment
+http:
+    public:
+        certificates:
+            - cert-1
+            - cert-2
+    private:
+      ingress:
+        vpc: true
+`,
+			wantedStruct: &Environment{
+				Workload: Workload{
+					Name: aws.String("prod"),
+					Type: aws.String("Environment"),
+				},
+				EnvironmentConfig: EnvironmentConfig{
+					HTTPConfig: EnvironmentHTTPConfig{
+						Public: PublicHTTPConfig{
+							Certificates: []string{"cert-1", "cert-2"},
+						},
+						Private: privateHTTPConfig{
+							Ingress: RelaxedIngress{VPCIngress: aws.Bool(true)},
+						},
+					},
+				},
+			},
+		},
+		"unmarshal with new and old private http fields": {
+			inContent: `name: prod
+type: Environment
+http:
+    public:
+        certificates:
+            - cert-1
+            - cert-2
+    private:
+      security_groups:
+        ingress:
+          from_vpc: true
+      ingress:
+        vpc: true
+`,
+			wantedStruct: &Environment{
+				Workload: Workload{
+					Name: aws.String("prod"),
+					Type: aws.String("Environment"),
+				},
+				EnvironmentConfig: EnvironmentConfig{
+					HTTPConfig: EnvironmentHTTPConfig{
+						Public: PublicHTTPConfig{
+							Certificates: []string{"cert-1", "cert-2"},
+						},
+						Private: privateHTTPConfig{
+							Ingress: RelaxedIngress{VPCIngress: aws.Bool(true)},
+							DeprecatedSG: DeprecatedALBSecurityGroupsConfig{
+								DeprecatedIngress: DeprecatedIngress{
+									VPCIngress: aws.Bool(true),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"unmarshal with new and old public http fields": {
+			inContent: `name: prod
+type: Environment
+http:
+    public:
+      certificates:
+        - cert-1
+        - cert-2
+      security_groups:
+        ingress:
+          restrict_to:
+            cdn: true
+      ingress:
+        cdn: true
+`,
+			wantedStruct: &Environment{
+				Workload: Workload{
+					Name: aws.String("prod"),
+					Type: aws.String("Environment"),
+				},
+				EnvironmentConfig: EnvironmentConfig{
+					HTTPConfig: EnvironmentHTTPConfig{
+						Public: PublicHTTPConfig{
+							Certificates: []string{"cert-1", "cert-2"},
+							DeprecatedSG: DeprecatedALBSecurityGroupsConfig{DeprecatedIngress: DeprecatedIngress{RestrictiveIngress: RestrictiveIngress{CDNIngress: aws.Bool(true)}}},
```

Comment:
nit: (it's difficult to indent using gh editor but let's not squash them into one line 🥺 )
```suggestion
							DeprecatedSG: DeprecatedALBSecurityGroupsConfig{
							  DeprecatedIngress: DeprecatedIngress{
							    RestrictiveIngress: RestrictiveIngress{
							      CDNIngress: aws.Bool(true)
							      }
							    }
							   },
```